File: app-ui/src/main/java/com/teewhydope/app/ui/login/model/LoginResponseUiModel.kt
package com.teewhydope.app.ui.login.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class LoginResponseUiModel(
    val userData: UserDataUiModel,
    val smartSaverTransactions: List<SmartSaverTransactionUiModel>
) : Parcelable

@Parcelize
data class SmartSaverTransactionUiModel(
    val amount: Long,
    val type: String,
    val narration: String,
    val dateCreated: String
) : Parcelable

@Parcelize
data class UserDataUiModel(
    val firstName: String,
    val smartSaverBalance: Long,
    val greenSaverBalance: Long,
    val fixedDepositBalance: Long,
) : Parcelable

File: iosApp/iosApp/ViewController/Login/Model/LoginResponseUiModel.swift
//
//  LoginResponseUiModel.swift
//  iosApp
//
//  Created by Animasahun Ibrahim on 17/03/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation

struct LoginResponseUiModel{
    let userData: UserDataUiModel
    let smartSaverTransactions: [SmartSaverTransactionUiModel]
}

struct SmartSaverTransactionUiModel{
    let amount: Int64
    let type: String
    let narration: String
    let dateCreated: String
}

struct UserDataUiModel {
    let firstName: String
    let smartSaverBalance: Int64
    let greenSaverBalance: Int64
    let fixedDepositBalance: Int64
}

